### ISSUE TITLE
[Feature] Calendar widget

### ISSUE THREAD

**Issue by areyabhishek** [appsmithorg/appsmith#2410]
## Summary

A calendar widget is a time grid. It'll be useful for showing events, schedules, etc. It's especially useful for teams that deal with deadlines like e-commerce companies, content creation companies, etc.

Features:
1. Ability to move events around
2. Ability to drill down to hour level, day level, week level, month level, year level
3. Creating events by selecting the time period 

<img width="1083" alt="Screen Shot 2020-12-31 at 7 45 33 PM" src="https://user-images.githubusercontent.com/30255708/103413728-c06ab000-4ba0-11eb-9711-77b76c972147.png">

<img width="1085" alt="Screen Shot 2020-12-31 at 7 44 20 PM" src="https://user-images.githubusercontent.com/30255708/103413671-94e7c580-4ba0-11eb-9841-95de28622cd1.png">

<img width="1081" alt="Screen Shot 2020-12-31 at 7 43 21 PM" src="https://user-images.githubusercontent.com/30255708/103413631-71bd1600-4ba0-11eb-83de-c96c06b8af9e.png">

References: 
https://fullcalendar.io/
https://github.com/nhn/tui.calendar
https://github.com/jquense/react-big-calendar

**Comment by momcilo-appsmith** [comment]
@riodeuno @somangshu Can we schedule a discussion around this widget?

**Comment by gabriel-lucas** [comment]
I think it would very valuable to have this widget integrated in appsmith

**Comment by Michael-sr** [comment]
+1

**Comment by orenab1** [comment]
Would've been great if this existed.. I'm building an app which MUST have it..

**Comment by somangshu** [comment]
Hey @orenab1, thanks for showing your interest here. This widget is not yet planned to be developed. However I have a demo [here](https://app.appsmith.com/applications/6138fbeddd7786286ddd4df1/pages/6138fbeddd7786286ddd4df3), which might let you create an alternate solution. Do check this out and let me know what are your thoughts, The context you provide will help us triage this issue in order to pick it up

**Comment by orenab1** [comment]
Hi, since I've written that post, I realized I don't actually need a
calendar.. I've also sticked with my Angular/.NET Core application..
But I'll definitely consider AppSmith next time.

Thanks!

**Comment by PaulMcF1987** [comment]
This is also something that I feel would be a great addition. Currently, I am using Appsmith alongside OpenProject as open project allows me to use a calendar to keep track of events etc, If I could move away from open project and have a calender widget on Appsmith this would make everything so much easier.

(CalDav/CardDav synchronization would also be great, lol)

I have looked at the sample app from @somangshu but I dont think that would cover what I am looking for as some tasks span over multiple days so the ability to have them displayed in a similar fashion to the below pic would be great

![image](https://user-images.githubusercontent.com/75368133/197489685-07d634d1-e7ee-40a0-90fd-12f675d41eb7.png)

**Comment by somangshu** [comment]
Hey @PaulMcF1987, thanks for providing us with the context; this widget is not yet on our roadmap and we dont plan on picking this up anytime soon. If you can help us understand your use case better we might be able to pull another solution with some other widgets; We probably can color code the date tiles to resemble the above UX.

cc @dilippitchika FYI

**Comment by PaulMcF1987** [comment]
@somangshu I run a transport company, I use Appsmith to plan jobs. A calendar function would allow me to easily see what jobs are due on what days,
Also for job sourcing, it would allow me to easily identify at a glance which days still have jobs which need allocated.

It is not as easy to taker the details off of a table, especially when trying to work at speed.

Also, I have made an app for my brother who is a driving instructor, he has multiple appointments every day. A calendar function would help him see, at a glance if he had any spaces left for that specific day. Currently I have added a filter to the table to filter each day, however if trying to find a free day for a pupil to book a lesson, it can take a while as he can be required to filter through multiple days.
If a calendar was available, a glance should be enough to see where a space in the diary occurs and would save him a lot of time.

**Comment by somangshu** [comment]
I understand @PaulMcF1987, Thanks for explaining the use case, I see how you wish to utilize the calendar widget and it makes sense as well. 
However there could be a couple of other UX which can be thought of here for making this work well right now;
One of the pain points you mentioned was not being able to identify the days that need allocation. In that case another table / list showing the same in a chronological order might do the job. I infer from your explanation that your schema already can fit this UX. 

Meanwhile, we will think about what kind of Impact this makes on the rest of our users and re-evaluate if this should be re-prioritized 

cc @dilippitchika Any better ideas?

**Comment by PaulMcF1987** [comment]
@somangshu I already have a few different queries which will arrange the table in different ways in order to make things a bit easier. But it is still no where near as easy as what a calendar widget would offer.

I genuinely believe that a calendar widget would be something that most/all businesses could benefit from.

All businesses out there require some sort of schedule and at present, whilst they may be using appsmith, they are more than likely going to have some sort of calendar/diary software which they use.
Currently, I use Openproject alongside appsmith

However, the ability to condense this into one app and get rid of the software that they are using for their time management and add this into appsmith I genuinely believe that would be a massive advantage to a lot of people businesses, and could be a good selling point for the software

**Comment by somangshu** [comment]
Agreed @PaulMcF1987, we shall look into this requirement; Please subscribe to the issue to get notifications for any updates we leave here

**Comment by GreenFlux** [comment]
another user in Discord is requesting this feature. 
https://discord.com/channels/725602949748752515/1087360531846942820

**Comment by somangshu** [comment]
@GreenFlux can you document the use case for us here?

**Comment by GreenFlux** [comment]
@somangshu , from the user on Discord:
> Our use case is mainly displaying timestamped data. For example there is a form that users can use to schedule events, but no way to have a more "natural" way to view them.
So the calendar widget would be purely for visualizing things, no CRUD features needed (for us at least)
Tooljet [calendar widget](https://docs.tooljet.com/docs/widgets/calendar/) provides everything we need. Tooljet in general is missing some features that make it worse, like code formatting, etc.

**Comment by Nikhil-Nandagopal** [comment]
We recently added a custom widget where you can create a custom calendar widget.
Check out the post below where you can fork and try out the widget
https://community.appsmith.com/template/calendar
